parent/pom.xml

```diff
@@ -312,6 +312,11 @@
                 <artifactId>commons-math3</artifactId>
                 <version>3.2</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-lang3</artifactId>
```

Comment:
Not too sure about pulling in commons-lang3 just for the use of `Pair`.  It has so much overlap with Guava that including both could get confusing.  I'm not knocking its use completely, questioning whether it's the best approach here.  For example you created the static method `DefaultCompactionControlSource.getStashTimesWithTupleKeys()` which gets called to in numerous places to handle the pair logic.  Would it be cleaner to create a new class for this purpose which would encapsulate the Key->String-Key serialization and deserialization as well as serve as the globally unique identifier for a compaction control?